src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SpecializedTasks.cs

```diff
@@ -26,6 +26,10 @@ internal static class SpecializedTasks
         public static Task<T?> AsNullable<T>(this Task<T> task) where T : class
             => task!;
 
+        [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "This is a Task wrapper, not an asynchronous method.")]
+        public static ValueTask<T?> AsNullable<T>(this ValueTask<T> task)
+            => task!;
```

Comment:
i feel like this should be constrained to just `T : class`.  But maybe i could be convinced otherwise.  the .AsNullable sounds really confusing if T was a value type, and you didn't get any difference (as opposed to gettint `Nullable<T>` back).

Reply:
➡️ Amended the commit to make this change